.github/workflows/ci.yml

```diff
@@ -0,0 +1,101 @@
+name: CI
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened]
+
+concurrency:
+  group: ci-${{ github.sha }}
+  cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    if: ${{ contains(fromJSON('["synchronize", "opened", "reopened"]'), github.event.action) }}
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # tag=v3.3.0
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - uses: ./github-actions/setup-bazel-remote-exec
+      - uses: actions/cache@v3
+        with:
+          path: |
+            .yarn/cache
+            .yarn/install-state.gz
+          key: v1-${{hashFiles('yarn.lock')}}
+      - run: yarn install --immutable
+      - run: yarn ng-dev format changed --check ${{ github.event.pull_request.base.sha }}
+      - run: yarn ng-dev commit-message validate-range ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }}
+      - run: yarn lint
+      - run: yarn tsc -p tsconfig.json
+
+  test:
+    if: ${{ contains(fromJSON('["synchronize", "opened", "reopened"]'), github.event.action) }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # tag=v3.3.0
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: ./github-actions/setup-bazel-remote-exec
+      - uses: actions/cache@v3
+        with:
+          path: |
+            .yarn/cache
+            .yarn/install-state.gz
```

Comment:
We should extract the caching into a common action, like we do in components: https://github.com/angular/components/blob/main/.github/actions/yarn-install/action.yml

This makes cache key maintenance much more central, easier, and removes bloat in individual job steps

Reply:
Definitely, I wanted to start with leaving things in place during this testing/migration in dev-infra and then extract before we roll wider.